scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3224,10 +3224,11 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     }
     val className = typeName()
     rejectModCombination[Mod.Final, Mod.Sealed](mods, s"class $className")
-    // TODO:
-    // if (ofCaseClass && token.isNot[LeftParen])
-    //  syntaxError(token.offset, "case classes without a parameter list are not allowed;\n"+
-    //                             "use either case objects or case classes with an explicit `()' as a parameter list.")
+    if (mods.has[Mod.Case] && token.isNot[LeftParen] && !dialect.allowCaseClassWithoutParameterList) {
```

Comment:
Let's start  this condition with `dialect.allow...`, that makes it easier to read.